homeassistant/components/nuki/config_flow.py

```diff
@@ -77,6 +83,7 @@ async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowRes
                 vol.Required(CONF_HOST, default=discovery_info.ip): str,
                 vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
                 vol.Required(CONF_TOKEN): str,
+                vol.Required(CONF_ENCRYPT_TOKEN, default=True): bool,
```

Comment:
Why is it required here and for the other schemas not?

As a secure connection is recommended, I would try to encrypt the token directly, and only if `async_step_validate` fails, show the user the option to disable the encryption. What do you think?

Reply:
I think we could do that. I'll implement that tomorrow.

Reply:
I've changed it now, so that if connection to the Nuki Bridge fails (regardless of the error code) it will show the form again with the option to disable the encrypted token.